posydon/popsyn/synthetic_population.py

```diff
@@ -197,36 +226,56 @@ def parse(self, S1_state=None, S2_state=None, binary_state=None,
             Read the POSYDON binary population in chuncks to prevent OFM error.
 
         """
+        
+        # if the user provided a single string instead of a list of strings
+        if type(path_to_data) is str and ('.h5' in path_to_data):
+            path_to_data = [path_to_data]
+        
+        # catch the case where the user did not provide a path to data     
+        if not (isinstance(path_to_data, list) and '.h5' in path_to_data[0]):
+            raise ValueError('You did not provide a valid path_to_data!')
 
+        
         df_sel = pd.DataFrame()
         df_sel_oneline = pd.DataFrame()
         count = 0
         tmp = 0
         if self.verbose:
-            print('Binary count with (S1_state, S2_state, binary_state, binary_event) equal')
-            print(f'to ({S1_state}, {S2_state}, {binary_state}, {binary_event})')
+            print('Binary count with (S1_state, S2_state, binary_state, binary_event, step_name) equal')
+            print(f'to ({S1_state}, {S2_state}, {binary_state}, {binary_event}, {step_name})')
             if invert_S1S2:
-                print(f'and ({S2_state}, {S1_state}, {binary_state}, {binary_event})')
-        for k, file in enumerate(self.path_to_data):
+                print(f'and ({S2_state}, {S1_state}, {binary_state}, {binary_event}, {step_name})')
+        for k, file in enumerate(path_to_data):
             df_sel_met = pd.DataFrame()
             sel_met = []
+            last_binary_df = None
+            
             # read metallicity from path
             met = float(file.split('/')[-1].split('_Zsun')[0])*Zsun
             simulated_mass_for_met = 0.
-            # TODO: handle binaries at the edge case of the chuncks
+            
             for i, df in enumerate(pd.read_hdf(file,  key='history', chunksize=chunksize)):
-
-                logic = self.apply_logic(df, S1_state=S1_state,
-                                    S2_state=S2_state,
-                                    binary_state=binary_state,
-                                    binary_event=binary_event,
-                                    invert_S1S2=invert_S1S2)
-
+                
+                if last_binary_df is not None:
+                    df = pd.concat([last_binary_df, df])
+                    last_binary_df = None
```

Comment:
Why do we need this? We set it here to None and in the next code line it gets a new assignment nevertheless.

Reply:
Since `pd.concat` at the moment is fine with a None input it works.